src/Physics/HadronTransport/INukeUtils2025.cxx

```diff
@@ -0,0 +1,1934 @@
+
+      double E0 = TMath::Power(A,0.2)*12.;
+      if (INukeMode=="hN2025"){if(ke<E0){sigtot=0.0;}}  //empirical - needed to cut off large number of low energy nucleons
+      //      LOG("INukeUtils",pDEBUG) "sigtot for proton= " << sigtot << "; KE= " << ke << "; E0= " << E0;
+    }
+  else if (pdgc == kPdgNeutron)
+    {
+      sigtot = fHadroData2025 -> XSecPn_Tot()   -> Evaluate(ke)*ppcnt;
+      sigtot+= fHadroData2025 -> XSecNn_Tot()   -> Evaluate(ke)*(1-ppcnt);
+      double E0 = TMath::Power(A,0.2)*12.;
+      if (INukeMode=="hN2025"){if(ke<E0){sigtot=0.0;}}  //empirical - needed to cut off large number of low energy nucleons
+      //      LOG("INukeUtils",pDEBUG) "sigtot for neutron= " << sigtot << "; KE= " << ke;
+    }
+  else if (pdgc == kPdgKP)
+    { sigtot = fHadroData2025 -> XSecKpN_Tot()  -> Evaluate(ke);
+      // this factor is used to empirically get agreement with tot xs data, justified historically.
+      sigtot*=1.1;}
+  else if (pdgc == kPdgGamma)
+    { sigtot = fHadroData2025 -> XSecGamp_fs()  -> Evaluate(ke)*ppcnt;
+      sigtot+= fHadroData2025 -> XSecGamn_fs()  -> Evaluate(ke)*(1-ppcnt);}
+  else {
+     return 0;
+  }
+
+  // the xsection splines in INukeHadroData return the hadron x-section in
+  // mb -> convert to fm^2
+  sigtot *= (units::mb / units::fm2);
+
+  // avoid defective error handling
+  //if(sigtot<1E-6){sigtot=1E-6;}
+
+  if (xsecNNCorr and is_nucleon)
+    sigtot *= INukeNucleonCorr::getInstance()->
+      getAvgCorrection (rho, A, p4.E() - PDGLibrary::Instance()->Find(pdgc)->Mass());   //uses lookup tables
+
+  // avoid defective error handling
+  if(sigtot<1E-6){sigtot=1E-6;}
+
+  // compute the mean free path
+  double lamda = 1. / (rho * sigtot);
+
+  // exits if lamda is InF (if cross section is 0)
+  if( ! TMath::Finite(lamda) ) {
+     return -1;
+  }
+
+/*
+  LOG("INukeUtils", pDEBUG)
+     << "sig_total = " << sigtot << " fm^2, rho = " << rho
+     << " fm^-3  => mfp = " << lamda << " fm.";
+*/
+  return lamda;
+}
+//____________________________________________________________________________
+double genie::utils::intranuke2025::MeanFreePath_Delta(
+   int pdgc, const TLorentzVector & x4, const TLorentzVector & p4, double A)
+{
+//
+// **test**
+//
+
+// Calculate the mean free path (in fm) for Delta's in a nucleus
+//
+// Inputs
+//  pdgc : Hadron PDG code
+//  x4   : Hadron 4-position in the nucleus coordinate system (units: fm)
+//  p4   : Hadron 4-momentum (units: GeV)
+//  A    : Nucleus atomic mass number
+//
+  bool is_deltapp = (pdgc==kPdgP33m1232_DeltaPP);
+  if(!is_deltapp) return 0.;
+
+  // get the nuclear density at the current position
+  double rnow = x4.Vect().Mag();
+  double rho  = A * utils::nuclear::Density(rnow,(int) A);
+
+  // the Delta+N->N+N cross section will be evaluated within the range
+  // of the input spline and assumed to be const outside that range
+  double ke = (p4.Energy() - p4.M()) / units::MeV;  // kinetic energy in MeV
```

Comment:
The next two lines confuse me.   Are they intended to restrict ke to [0,1500]?   If so, I think the `Max` and `Min` are swapped.   Any starting `ke` less than `1500` will get pushed to `1500`, then take the minimum of `0` and `1500` and get `0`.   

Actually, working through the logic, I think one _always_ gets `0`.